picard/ui/coverartbox.py

```diff
@@ -189,16 +189,13 @@ def on_remote_image_fetched(self, url, data, reply, error, fallback_data=None):
             log.warning("Can't load remote image with MIME-Type %s", mime)
             if fallback_data:
                # Tests for image format obtained from file-magic
-               if fallback_data[:2]=='\xff\xd8':
-                   mime = 'image/jpeg'
-               elif fallback_data[:8]=='\x89PNG\x0d\x0a\x1a\x0a':
-                   mime = 'image/png'
-               else:
-                   mime = None
-
-               if mime:
-                   log.debug("Trying the dropped %s data", mime)
+               try:
+                   mime = imageinfo.identify(fallback_data)[2]
+                   log.warning("Trying the dropped %s data", mime)
                    self.load_remote_image(url, mime, fallback_data)
+               except:
```

Comment:
This catchall except isn't great imho.
The error message should be more informative.